src/lib_ccaligner/grammar_tools.cpp

```diff
@@ -17,9 +17,274 @@ static int systemGetStatus(const char* command) {
+
+		if (rv != 0)
+		{
+			FATAL(EXIT_FAILURE, "Something went wrong while creating dictionary!");
+		}
+	}
+
+}
+
+void CreateTempDirectories()
+{
+	//create temporary directories in case they don't exist
+	LOG("Creating temporary directories at tempFiles/");
+
+#ifndef WIN32
+	int rv = systemGetStatus("mkdir -p tempFiles/corpus tempFiles/dict tempFiles/vocab tempFiles/fsg tempFiles/lm");
+#else
+	int rv = systemGetStatus("if not exist tempFiles\\ mkdir tempFiles\\corpus tempFiles\\dict tempFiles\\vocab tempFiles\\fsg tempFiles\\lm");
+#endif
+
+	if (rv != 0)
+	{
+		FATAL(EXIT_FAILURE, "Unable to create directory tempFiles/ : %s", strerror(errno));
+	}
+
+	LOG("Directories created successfully!");
+}
+
+std::string getFileData(std::string _fileName)           //returns whole read file
+{
+	std::ifstream infile(_fileName);
+	std::string allData = "";
+	std::string line;
+	while (std::getline(infile, line))
+	{
+		std::istringstream iss(line);
+		allData += line + "\n";
+	}
+	return allData;
+}
+
+bool generate(std::string transcriptFileName, grammarName name) //Generate Grammar from text files.
+{
+	std::string transcript = getFileData(transcriptFileName);
+
+	bool generateQuickDict = false, generateQuickLM = false;
+	int rv;
+
+	if (name == quick_dict)
+	{
+		generateQuickDict = true;
+		name = complete_grammar;
+	}
+
+	else if (name == quick_lm)
+	{
+		generateQuickLM = true;
+		name = complete_grammar;
+	}
+
+	CreateTempDirectories();
+
+	std::ofstream corpusDump, phoneticCorpusDump, fsgDump, vocabDump, dictDump, logDump;
+
+	//setting exceptions to be thrown on failure
+	corpusDump.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+	fsgDump.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+	vocabDump.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+	dictDump.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+	phoneticCorpusDump.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+	logDump.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+
+	try
+	{
+		logDump.open("tempFiles/grammar.log", std::ios::binary);
+		logDump.close();
+	}
+	catch (std::system_error& e)
+	{
+		FATAL(EXIT_FAILURE, e.code().message().c_str());
+	}
+
+	if (name == corpus || name == complete_grammar)
+	{
+		std::cout << "Creating Corpus : tempFiles/corpus/corpus.txt\n";
+
+		try
+		{
+			corpusDump.open("tempFiles/corpus/corpus.txt", std::ios::binary);
+			corpusDump.close();
+		}
+		catch (std::system_error& e)
+		{
+			FATAL(EXIT_FAILURE, e.code().message().c_str());
+		}
+	}
+	
+	if (name == phone_lm || name == complete_grammar)
+	{
+		std::cout << "Creating Phonetic Corpus : tempFiles/corpus/phoneticCorpus.txt\n";
+
+		try
+		{
+			phoneticCorpusDump.open("tempFiles/corpus/phoneticCorpus.txt", std::ios::binary);
+			phoneticCorpusDump.close();
+		}
+
+		catch (std::system_error& e)
+		{
+			FATAL(EXIT_FAILURE, e.code().message().c_str());
+		}
+
+	}
```

Comment:
The lines from 146 till here also looks like the same shared between the two generate functions. Can you look at them as well?

Reply:
Sure. Will do asap.